### PR TITLE
Replace the PMIX_DECLSPEC declaration with PMIX_EXPORT for consistency

### PR DESCRIPTION
**Pull request by rhc54**
Signed-off-by: Ralph Castain <rhc@open-mpi.org>

### CODE REVIEW

**Comment by hppritcha** [comment]
@jhursey can you review asap?  We'd like to get this in to a 2.0.4rc2 and turnaround by Wednesday.
I'd like at least one batch of MTT runs against it.

**Comment by jsquyres** [comment]
bot:mellanox:retest

File: opal/mca/pmix/pmix112/pmix/include/pmix/autogen/pmix_config_bottom.h [hunk 1]
```diff
@@ -343,26 +343,10 @@
...
-# endif
-#else
-# define PMIX_EXPORT
-#endif
-#endif
-
-#ifndef PMIX_DECLSPEC
-#define PMIX_DECLSPEC
-#  if PMIX_C_HAVE_VISIBILITY
+#if PMIX_C_HAVE_VISIBILITY
```
Comment (jjhursey) [inline opal/mca/pmix/pmix112/pmix/include/pmix/autogen/pmix_config_bottom.h]: Minor nit: But I would add the redefinition protection before this out of "abundance of caution".
```diff
+ #ifndef PMIX_EXPORT
  #if PMIX_C_HAVE_VISIBILITY
  #    define PMIX_EXPORT           __pmix_attribute_visibility__("default")
  #else
  #    define PMIX_EXPORT
  #endif
+ #endif
```

It's minor enough that I'm not willing to hold up approval for this PR.

**Comment by jsquyres** [comment]
@Di0gen @jladd-mlnx @artpol84 Mellanox Jenkins is failing on this PR and some others due to what looks like a false failure.  Can you please look into what is happening?

```
11:09:34 --> Created destination specfile: /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/rpms/SPECS/openmpi-2.0.4rc1.spec
11:09:34 --> Building the Open MPI SRPM
11:09:34 --> rpmbuild --define 'mflags -j8' --define '_source_filedigest_algorithm md5' --define '_binary_filedigest_algorithm md5' --define '_topdir /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/rpms' --define 'dist %{nil}' -bs /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/rpms/SPECS/openmpi-2.0.4rc1.spec
11:09:34 error: Unable to write temp header
11:09:34 *** FAILURE BUILDING SRPM!
11:09:34 Aborting
11:09:34 + jenkins_cleanup
```

**Comment by jsquyres** [comment]
bot:mellanox:retest

**Pull request merged**